### PR TITLE
CotE - Seppun hidden guard

### LINKED ISSUE #2791: CotE - Seppun Hidden Guard
shg

### PR DESCRIPTION
**Pull request by Matthew-Griffith**
closes #2791 

here is the early version of the card class. it looks like it mostly working expect the random discard part of the card.

### CODE REVIEW

File: server/game/cards/06-CotE/SeppunHiddenGuard.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+                    card.isUnique() && card.controller === context.player && card.location === Locations.PlayArea)
+                )
+            },
+            cost: AbilityDsl.costs.sacrificeSelf(),
+            effect: 'cancel the effects of {1}',
+            effectArgs: context => context.event.card,
+            handler: context => {
+                context.cancel();
+                // AbilityDsl.actions.discardAtRandom();
+                context.game.applyGameAction(context, { discardAtRandom: context.player.opponent });
```
Comment (jeremylarner) [inline server/game/cards/06-CotE/SeppunHiddenGuard.js]: In theory, you could use this ability to cancel your own ability, in which case you should discard a card.  I think this should target `context.event.context.player`

Comment (Matthew-Griffith) [reply server/game/cards/06-CotE/SeppunHiddenGuard.js]: good catch, I didn't think about that when I read the card

**Comment by jeremylarner** [comment]
By default, the client doesn't prompt you to cancel your own abilities.  You'll need to set that option for the test player if you want to test that functionality

**Comment by Matthew-Griffith** [comment]
oh thanks, I was really confused why that test was failing

**Pull request merged**